src/pathfinder/water_regions.cpp

```diff
@@ -316,15 +301,21 @@ WaterRegionPatchDesc GetWaterRegionPatchInfo(TileIndex tile)
 void InvalidateWaterRegion(TileIndex tile)
 {
 	if (!IsValidTile(tile)) return;
-	const int water_region_index = GetWaterRegionIndex(tile);
-	_water_regions[water_region_index].Invalidate();
+
+	auto invalidateRegion = [](TileIndex tile) {
```

Comment:
Other local lamdas in the source use snake case instead of camel. This may be worth revising.